config/hwloc_internal.m4

```diff
@@ -385,7 +411,12 @@ AC_DEFUN([HWLOC_SETUP_TESTS],[
 ###
 EOF
 
-    AC_CHECK_LIB([pthread], [pthread_self], [hwloc_have_pthread=yes])
+    # Check thread support.
+    AX_PTHREAD([
+	hwloc_have_pthread=yes;
+	AC_DEFINE([HAVE_PTHREAD], [1], [pthread support is here])
```

Comment:
Same as mentioned above:

* `AX_PTHREAD` could be problematic.  Are you just needing to see if you need to use `-pthread`?  If so, you might be able to steal Open MPI's pthread m4.
* Always define to 0 or 1.

Reply:
Done. See #https://github.com/open-mpi/hwloc/pull/359/commits/a7979a9c2aa3fd277a38c239432dcfb8db97202b